HopsanGUI/Widgets/LibraryWidget.cpp

```diff
@@ -754,6 +755,24 @@ void LibraryWidget::handleItemClick(QTreeWidgetItem *item, int column)
     }
 }
 
+void LibraryWidget::handleItemDoubleClick(QTreeWidgetItem *item, int column)
+{
+    if(item != nullptr &&
+        mItemToTypeNameMap.contains(item) &&
+        gpLibraryHandler->getEntry(mItemToTypeNameMap.find(item).value()).displayPath.startsWith(componentlibrary::roots::externalLibraries))
+    {
+        //Edit component source file
+        QString typeName = gpLibraryHandler->getModelObjectAppearancePtr(mItemToTypeNameMap.find(item).value())->getTypeName();
+        auto appearance = gpLibraryHandler->getModelObjectAppearancePtr(typeName);
```

Comment:
hmm no need to get the appearance ptr twice 
Also rather long chains, that may crash to program is something goes wrong somewhere. Lets say getModelObjectAppearance would return nullptr for some reason. I should not but maybe should always check pointer in non-performance-critical code.